File: c_benchmarks/float_ops.c
/* Float ops: sin/cos/sqrt over many points. */

#include <math.h>
#include <stdio.h>
#include <stdlib.h>

#include "common.h"

#define POINTS 2000000

void run(void) {
    double total = 0.0;
    for (long i = 0; i < POINTS; i++) {
        double x = (double)i * 0.0001;
        total += sin(x) * cos(x) + sqrt(x + 1.0);
    }
    if (total == 0.0) {
        fprintf(stderr, "unreachable\n");
        exit(1);
    }
}

File: benchmarks/float_ops.py
"""Float ops: sin/cos/sqrt over many points. Pure-Python interpreter stress."""

import math

POINTS = 2_000_000


def run() -> None:
    total = 0.0
    for i in range(POINTS):
        x = i * 0.0001
        total += math.sin(x) * math.cos(x) + math.sqrt(x + 1.0)
    # Prevent dead-code elimination
    if total == 0:
        raise RuntimeError("unreachable")
